packages/react/src/components/internal/CalendarRanges/CalendarRanges.tsx

```diff
@@ -0,0 +1,66 @@
+import type { RangeCalendarState } from '@react-stately/calendar';
+import * as React from 'react';
+import { cx } from '../../../styles';
+import { useStyles } from './CalendarRanges.styles';
+import { ListBox, ListBoxItem } from '../../ListBox';
+import { Selection } from '@react-types/shared';
+import { RangeValue } from '../../CalendarRange';
+import { CalendarDate } from '@internationalized/date';
+import { getDefaultRanges } from './defaultDefinedRanges';
+
+interface DefinedRange {
+  key: string;
+  label: string;
+  value: RangeValue<CalendarDate>;
+}
+interface CalendarRangesProps {
+  /**
+   * The calendar state.
+   */
+  state: RangeCalendarState;
```

Comment:
It may be a little more descriptive to change this prop too: `onRangeChange` and pass the handler (handleRangeChange) from the parent.  What do you think @bclark-p44?

Reply:
I like that idea!

Reply:
sure guys

Reply:
done